### DISCUSSION TITLE
Missing so many features for SONOFF TRVZB

### DISCUSSION THREAD

**Discussion by igittigitt** [Koenkk/zigbee2mqtt#19669] [Devices]
I've recently bought a Sonoff Zigbee TRV, but besides the thermostats features it found there is only a limited set of them available in Z2M integration of Home-Assistant. Why is it that way and how to extend this?

**Comment by lux73** [comment]
sorry, but it's known that the ZHA implementation from HA is very limited

you could try install plain z2m if theres more features

if you not like this idea - only way to ask @ HA Forum

**Comment by igittigitt** [comment]
I see, and they send me here...

**Reply by lux73** [reply]
hm, thats strange - afaik ZHA is independant from z2m, they share some code base...

then you have to deal with plain z2m installation

**Reply by igittigitt** [reply]
Yes, really strange. HA implements Z2M and i think that part is genuine and has nothing special for HA. Only the interface between Z2M and HA, but then i would expect to see the missing entities at least in the Z2M menus.

When i look here https://www.zigbee2mqtt.io/devices/TRVZB.html it lists only a few entities for the TRVZB, but i know that the TRV can do a lot more. It hat local calibration, child-lock, open-window detection, weekly programs, etc.
